src/controllers/SelectYearController.ts

```diff
@@ -0,0 +1,150 @@
+import { loggerInstance } from 'app/middleware/Logger';
+import { PenaltyReferenceRouter } from 'app/middleware/PenaltyReferenceRouter';
+import { Appeal } from 'app/models/Appeal';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { createPenaltyRadioButton } from 'app/models/components/PenaltyRadioButton';
+import { createSchema } from 'app/models/fields/PenaltyChoice.schema';
+import { APPLICATION_DATA_UNDEFINED, SESSION_NOT_FOUND_ERROR } from 'app/utils/CommonErrors';
+import { PENALTY_DETAILS_PAGE_URI, REVIEW_PENALTY_PAGE_URI, SELECT_YEAR_PAGE_URI } from 'app/utils/Paths';
+import { ValidationResult } from 'app/utils/validation/ValidationResult';
+
+const template = 'select-the-year';
+
+const navigation = {
+    previous(): string {
+        return PENALTY_DETAILS_PAGE_URI;
+    },
+    next(): string {
+        return REVIEW_PENALTY_PAGE_URI;
+    },
+    actions: (changeMode: boolean) => {
+        return {
+            continue: changeMode ? 'action=continue&cm=1' : 'action=continue'
+        };
+    }
+};
+
+@provide(Processor)
+class Processor implements FormActionProcessor {
+    public process(request: Request): void | Promise<void> {
+
+        const session = request.session;
+
+        if (!session) {
+            throw SESSION_NOT_FOUND_ERROR;
+        }
+
+        const appData: ApplicationData | undefined = session.getExtraData(APPLICATION_DATA_KEY);
+
+        if(!appData) {
+            throw APPLICATION_DATA_UNDEFINED;
+        }
+
+        const penaltySelected: string = request.body.selectPenalty;
+
+        appData.appeal.penaltyIdentifier.penaltyReference = penaltySelected;
+
+        session.setExtraData(APPLICATION_DATA_KEY, appData);
+    }
+
+}
+
+export const errorMessage: string = 'Select the penalty you want to appeal';
+
+// tslint:disable-next-line: max-classes-per-file
+@controller(SELECT_YEAR_PAGE_URI, SessionMiddleware, AuthMiddleware, PenaltyReferenceRouter)
+export class SelectYearController extends SafeNavigationBaseController<any> {
+
+    public static PENALTY_EXPECTED_ERROR: string = 'Penalty object expected but none found';
+
+    constructor() {
+        super(
+            template,
+            navigation,
+            new FormValidator(Joi.object({
+                selectPenalty: createSchema(errorMessage)
+            })),
+            undefined,
+            [Processor]
+        );
+    }
+
+    public prepareViewModelFromAppeal(appeal: Appeal): Record<string, any> & any {
+
+        const penaltyList: PenaltyList | undefined = appeal.penaltyIdentifier.penaltyList;
+
+        if (!penaltyList || !penaltyList.items) {
+            throw new Error(SelectYearController.PENALTY_EXPECTED_ERROR);
+        }
+
+        return { penalties: penaltyList.items.map(createPenaltyRadioButton) };
+    }
+
+    protected getExtraActionHandlers(): Record<string, FormActionHandler | FormActionHandlerConstructor> {
```

Comment:
Not sure I understand why you needed a custom action handler

Reply:
![image](https://user-images.githubusercontent.com/54070318/84752944-5567dc00-afb6-11ea-9ef3-eb3aa18a5e60.png)

Base Controller:
![image](https://user-images.githubusercontent.com/54070318/84753020-657fbb80-afb6-11ea-8d67-2460cdd383e1.png)

Reply:
If I don't include the view model when there is an error, the radio buttons will not populate with data because the assumption was made that the request body will contain the necessary data. Unfortunately when you submit the radio button form you only get the data for the selected radio button

Reply:
thanks for the explanation